### DISCUSSION TITLE
strange timing distribution

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#23216] [Q&A] [answered]
This is what happens when I play normally
![{$A$V6ZZQ_A18L3DR8RXQS](https://user-images.githubusercontent.com/34775378/232290585-48dc9a97-e6f0-4d92-b57f-c7fb09bab91c.png)

This should not occur under normal circumstances
However, this situation seems to happen often when I limit the frame rate to ~~vsync~~ under windowed mode
So I'm wondering, is the input ignoring something? I haven't had this problem before.

settings: D3D11 windowed multithreading ~~v-sync(60hz)~~

need set vsync first before start lazer

osr file
[cdwcgt playing Wakeshima Kanon - killy killy JOKER (TV Size) (A r M i N) [schoolboy_s Expert] (2023-04-16_18-35).zip](https://github.com/ppy/osu/files/11241828/cdwcgt.playing.Wakeshima.Kanon.-.killy.killy.JOKER.TV.Size.A.r.M.i.N.schoolboy_s.Expert.2023-04-16_18-35.zip)

[logs.zip](https://github.com/ppy/osu/files/11241818/logs.zip)
2023.403.1

**Comment by peppy** [comment, marked as answer]
Frames are processed every 4.2 ms at your settings. This looks correct.

**Reply by cdwcgt** [reply]
But shouldn't the input be handled by the input thread? Will it still be limited by the draw thread (fps)?

**Reply by cdwcgt** [reply]
More importantly, I have seen people claim that this kind of replay is played with cheating because people cannot do it. If this situation is common, then it will be a big problem.

**Reply by integerrr** [reply]
Also potentially related to your keyboards polling rate

**Reply by peppy** [reply]
> More importantly, I have seen people claim that this kind of replay is played with cheating because people cannot do it. If this situation is common, then it will be a big problem.

people are stupid

**Reply by Theighlin** [reply]
> Will it still be limited by the draw thread (fps)?

Those ms are update thread i believe.

**Reply by cdwcgt** [reply]
Then why does it also happen when I use a higher limiter, and it comes to normal after using full screen mode (As long as you switch back to window mode without restarting, it is still normal).

also, I mentioned this question has not come up before
